src/TemplateString.php

```diff
@@ -0,0 +1,174 @@
+<?php
+
+namespace ipl\Html;
+
+use Exception;
+
+/**
+ * Render {{#mustache}}Mustache{{/mustache}}-like string from {@link ValidHtml} element arguments
+ *
+ * # Example Usage
+ * ```
+ * $info = new TemplateString(
+ *     'Follow the %s for more information on %s {{#mustache}}Mustache{{/mustache}}',
+ *     [
+ *         new Link('doc/html', 'HTML documentation'),
+ *         Html::tag('strong', 'HTML elements'),
+ *        [
+ *        'mustache' => Html::tag('span')
+ *        ]
+ *     ]
```

Comment:
That example doesn't quite show the advantage of this template functionality. Use this instead:

```suggestion
* $info = TemplateString::create(
*      'Follow the {{#doc}}HTML documentation{{/doc}} for more information on {{#strong}}HTML elements{{/strong}}',
*      [
*          'doc'    => new Link(null, 'doc/html'),
*          'strong' => Html::tag('strong')
*      ]
```